pkg/volume/csi/csi_attacher.go

```diff
@@ -220,14 +220,18 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 			klog.Error(log("attacher.VolumesAreAttached missing volume.Spec"))
 			return nil, errors.New("missing spec")
 		}
-		source, err := getCSISourceFromSpec(spec)
+		pvSrc, err := getPVSourceFromSpec(spec)
 		if err != nil {
-			klog.Error(log("attacher.VolumesAreAttached failed: %v", err))
+			attached[spec] = false
```

Comment:
does this need to be explicitly set?

Reply:
Yes, if not it creates a bug that only shows up when there is an error where the `len(attached) != len(specs)`. It's bug that showed up during testing.